tests/sklearn/test_sklearn_models.py

```diff
@@ -1139,6 +1162,76 @@ def check_load_fitted_sklearn_linear_models(model_class, n_bits, x, y, check_flo
     )
 
 
+def check_rounding_consistency(
+    model,
+    n_bits,
+    x,
+    y,
+    predict_method,
+    metric,
+    is_weekly_option,
+):
+    """Test that Concrete ML without and with rounding are 'equivalent'."""
+
+    # Run the test with more samples during weekly CIs
+    if is_weekly_option:
+        fhe_test = get_random_samples(x, n_sample=5)
+
+    # Check that rounding is enabled
+    rounding_enabled = os.getenv("USE_ROUNDING") == "True"
+    assert rounding_enabled
+
+    # Fit and compile with rounding enabled
+    circuit_with_rounding = fit_and_compile(model, x, y)
+
+    rounded_predict_quantized = predict_method(x, fhe="disable")
+    rounded_predict_simulate = predict_method(x, fhe="simulate")
+
+    # Compute the FHE predictions only during weekly CIs
+    if is_weekly_option:
+        rounded_predict_fhe = predict_method(fhe_test, fhe="execute")
+
+    # For models that has less than 9 bits of quantization, we repeat the same experiment, but this
+    # time with rounding disabled
+    if n_bits <= 9:
```

Comment:
ok so based on what you explained in `test_rounding_consistency` , that means the stage 2 rounding feature is actually never tested right ? we are not able to be sure that it's correct (since we are not able to compile without rounding) ! 

if that's correct, I'm not sure why we would add this rounding feature in the second stage then !

Reply:
how did we assessed that rounding in the second stage worked in our experiments ? couldn't this test be similar to what has been done then ?

Reply:
btw if we keep the test this say I would say it's better to have a `use_rounding` parameter (or similar) in this `check_rounding_consistency` and use `if use_rounding:` here ! this `n_bits <= 9` is similar to what you do in `check_rounding_consistency` to better to directly initialize `use_rounding` there so that if we ever change this "threshold value" one day we limit the risk of forgetting to change this one as well !

Reply:
> if that's correct, I'm not sure why we would add this rounding feature in the second stage then !

Because it's faster than without rounding

If the user wants to use large data sets and high-precision bit widths, this execution is faster for him.

Reply:
yes but how can we make sure that it's the right computations ? it can be faster but wrong 😅 I'm just saying that currently we are not testing that